toolchain/README.md

```diff
@@ -0,0 +1,204 @@
+# Toolchain
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+The toolchain represents the production portion of Carbon. At a high level, the
+toolchain's top priorities are:
+
+-   Correctness.
+-   Quality of generated code, including its performance.
+-   Compilation performance.
+-   Quality of diagnostics for incorrect or questionable code.
+
+The main compiler is `//toolchain/driver:carbon`. When compiling, the current
+flow of data is:
+
+1. Load the file into a [SourceBuffer](source/source_buffer.h).
+2. Lex a SourceBuffer into a [TokenizedBuffer](lexer/tokenized_buffer.h).
+3. Parse a TokenizedBuffer into a [ParseTree](parser/parse_tree.h).
+4. Transform a ParseTree into a [SemanticsIR](semantics/semantics_ir.h).
+5. This flow is still incomplete: code generation, using LLVM, is still
+   required.
+
+## Lexing
+
+The [TokenizedBuffer](lexer/tokenized_buffer.h) is the central point of lexing.
+
+The entire source buffer is converted into tokens before parsing begins. Tokens
+are referred to by an opaque handle, `TokenizedBuffer::Token`, which is
+represented as a 32-bit integer into the token array. The tokenized buffer can
```

Comment:
Minor wording alternative that focus more on the design and less on the implementation:

```suggestion
represented as a dense integer index into the buffer. The tokenized buffer can
```